multimap/src/test/java/org/infinispan/multimap/impl/DistributedMultimapListCacheTest.java

```diff
@@ -0,0 +1,130 @@
+import static org.infinispan.multimap.impl.MultimapTestUtils.PEPE;
+import static org.infinispan.multimap.impl.MultimapTestUtils.RAMON;
+import static org.testng.AssertJUnit.assertNotNull;
+import static org.testng.AssertJUnit.assertTrue;
+
+@Test(groups = "functional", testName = "distribution.DistributedMultimapListCacheTest")
+public class DistributedMultimapListCacheTest extends BaseDistFunctionalTest<String, Collection<Person>> {
+
+   protected Map<Address, EmbeddedMultimapListCache<String, Person>> listCluster = new HashMap<>();
+   protected boolean fromOwner;
+
+   public DistributedMultimapListCacheTest fromOwner(boolean fromOwner) {
+      this.fromOwner = fromOwner;
+      return this;
+   }
+
+   @Override
+   protected void createCacheManagers() throws Throwable {
+      super.createCacheManagers();
+
+      for (EmbeddedCacheManager cacheManager : cacheManagers) {
+         EmbeddedMultimapCacheManager multimapCacheManager = (EmbeddedMultimapCacheManager) EmbeddedMultimapCacheManagerFactory.from(cacheManager);
+         MultimapCacheManagerConfigurationBuilder builder = new MultimapCacheManagerConfigurationBuilder((GlobalConfigurationBuilder) null);
+         builder.addMultimap().name(cacheName);
+         MultimapCacheManagerConfiguration conf = builder.create();
+         AggregateCompletionStage<Void> definitions = CompletionStages.aggregateCompletionStage();
+         for (EmbeddedMultimapConfiguration c : conf.multimaps().values()) {
+            definitions.dependsOn(multimapCacheManager.defineConfiguration(c));
+         }
+         CompletionStages.join(definitions.freeze());
+         listCluster.put(cacheManager.getAddress(), multimapCacheManager.getMultimapList(cacheName));
+      }
+   }
+
+   @Override
+   protected SerializationContextInitializer getSerializationContext() {
+      return MultimapSCI.INSTANCE;
+   }
+
+   @Override
+   protected String[] parameterNames() {
+      return concat(super.parameterNames(), "fromOwner");
+   }
+
+   @Override
+   protected Object[] parameterValues() {
+      return concat(super.parameterValues(), fromOwner ? Boolean.TRUE : Boolean.FALSE);
+   }
+
+   @Override
+   public Object[] factory() {
+      return new Object[]{
+            new DistributedMultimapListCacheTest().fromOwner(false).cacheMode(CacheMode.DIST_SYNC).transactional(false),
+            new DistributedMultimapListCacheTest().fromOwner(true).cacheMode(CacheMode.DIST_SYNC).transactional(false),
+      };
+   }
+
+   @Override
+   protected void initAndTest() {
+      for (EmbeddedMultimapListCache list : listCluster.values()) {
+         assertThat(await(list.size(NAMES_KEY))).isEqualTo(0L);
+      }
+   }
+
+   protected EmbeddedMultimapListCache<String, Person> getMultimapCacheMember() {
+      return listCluster.
+            values().stream().findFirst().orElseThrow(() -> new IllegalStateException("Cluster is empty"));
+   }
+
+   public void testOffer() {
```

Comment:
I would expect to test duplicates and order

Reply:
Actually is there any reason the DIST and single node ones are testing different things?

Reply:
Not to test all, only that dist mode works at this point.
I wan to improve operations and testing with the implementation of the resp commands iteratively

Reply:
Sure, this seems like a good place to test the backend list implementation though. But I guess we can let it pass for now, but we need to make sure there is proper coverage for the various cases.

Reply:
I added some more tests here